src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -480,6 +510,53 @@ private ClusterDescription ProcessShardedChange(ClusterDescription clusterDescri
             return clusterDescription.WithServerDescription(args.NewServerDescription);
         }
 
+        private ClusterDescription ProcessStandaloneChange(ClusterDescription clusterDescription, ServerDescriptionChangedEventArgs args)
+        {
+            if (args.NewServerDescription.Type != ServerType.Standalone)
+            {
+                return RemoveServer(clusterDescription, args.NewServerDescription.EndPoint, "Server is not a standalone server.");
```

Comment:
If the one and only standalone is ever removed the cluster will never ever connect to anything again. The same thing happens if the last mongos is ever removed.

But apparently that's according to spec.